File: app/src/main/java/com/meli/pdesire/projectmeliaudioeffects/EngineFragment.java
package com.meli.pdesire.projectmeliaudioeffects;

import android.annotation.TargetApi;
import android.os.Build;
import android.os.Bundle;
import android.preference.Preference;
import android.preference.PreferenceFragment;

/**
 * Created by PDesire on 18.01.2016.
 */
public class EngineFragment extends PreferenceFragment {

    @TargetApi(Build.VERSION_CODES.HONEYCOMB)

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate (savedInstanceState);

        addPreferencesFromResource (R.xml.pref_engines);
        setHasOptionsMenu (true);

        Preference yume = findPreference("Yume");
        Preference meli = findPreference("Meli");


        yume.setOnPreferenceClickListener(new Preference.OnPreferenceClickListener () {
            @Override
            public boolean onPreferenceClick(Preference preference) {
                try {
                    String command = "su -c sh /system/Desire/Shells/Engines/Yume/Final/Final.sh";
                    Process p = Runtime.getRuntime ().exec (command);
                } catch (Exception e) {
                    e.printStackTrace ();
                }
                return false;
            }
        });

        meli.setOnPreferenceClickListener(new Preference.OnPreferenceClickListener () {
            @Override
            public boolean onPreferenceClick(Preference preference) {
                try {
                    String command = "su -c sh /system/Desire/Shells/Rage_Enable.sh";
                    Process p = Runtime.getRuntime ().exec (command);
                } catch (Exception e) {
                    e.printStackTrace ();
                }
                return false;
            }
        });
    }
}

File: app/src/main/java/com/meli/pdesire/projectmeliaudioeffects/EngineFragment.kt
package com.meli.pdesire.projectmeliaudioeffects

/**
 * Created by PDesire on 20.05.2017.
 */

import android.annotation.TargetApi
import android.os.Build
import android.os.Bundle
import android.preference.Preference
import android.preference.PreferenceFragment

class EngineFragment : PreferenceFragment() {

    @TargetApi(Build.VERSION_CODES.HONEYCOMB)
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        addPreferencesFromResource(R.xml.pref_engines)
        setHasOptionsMenu(true)

        val yume = findPreference("Yume")
        val meli = findPreference("Meli")


        yume.onPreferenceClickListener = Preference.OnPreferenceClickListener {
            try {
                val command = "su -c sh /system/Desire/Shells/Engines/Yume/Final/Final.sh"
                val p = Runtime.getRuntime().exec(command)
            } catch (e: Exception) {
                e.printStackTrace()
            }

            false
        }

        meli.onPreferenceClickListener = Preference.OnPreferenceClickListener {
            try {
                val command = "su -c sh /system/Desire/Shells/Rage_Enable.sh"
                val p = Runtime.getRuntime().exec(command)
            } catch (e: Exception) {
                e.printStackTrace()
            }

            false
        }
    }
}
